libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRouteDetector.java

```diff
@@ -52,10 +51,13 @@
   @Override
   public boolean isUserOffRoute(Location location, RouteProgress routeProgress, MapboxNavigationOptions options) {
 
+    if (checkDistanceRemaining(routeProgress)) {
+      return true;
+    }
+
     if (!validOffRoute(location, options)) {
       return false;
     }
-
     Point currentPoint = Point.fromLngLat(location.getLongitude(), location.getLatitude());
 
     // Get distance from the current step to future point
```

Comment:
What about extracting blocks of code into well-named (based on the comments) `private` methods?. We should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).